### ISSUE TITLE
field_reassign_with_default triggers inside `macro_rules!` macro

### ISSUE THREAD

**Issue by jplatte** [rust-lang/rust-clippy#8153]
### Summary

For a macro, it might not always be practical to replace a field reassign after (default) initialization with a struct literal. In the case of [this test](https://docs.rs/assign/1.1.1/src/assign/lib.rs.html#142-145), the only purpose of the macro being tested is providing struct-update like syntax but using assignments under the hood.

### Lint Name

field_reassign_with_default

### Reproducer

I tried this code:

```rust
use assign::assign; // version 1.1.1

#[derive(Default)]
struct SomeStruct {
    a: u32,
    b: Option<f32>,
    c: Option<u64>,
}

let res = assign!(SomeStruct::default(), {
    a: 5,
    b: None,
});
```

I saw this happen (condensed form of full output):

```
warning: field assignment outside of initializer for an instance created with Default::default()
note: consider initializing the variable with `tests::SomeStruct { b: assign!(SomeStruct::default(), {
              b,
              c: 1.into(),
          }), c: 1.into(), ..Default::default() }` and removing relevant reassignments
```

I expected to see this happen: No warnings

### Version

```text
rustc 1.59.0-nightly (e95e084a1 2021-12-19)
binary: rustc
commit-hash: e95e084a14870a718c712936ab5a8f8cd0159485
commit-date: 2021-12-19
host: x86_64-unknown-linux-gnu
release: 1.59.0-nightly
LLVM version: 13.0.0
```
